server/src/services/formation.service.ts

```diff
@@ -0,0 +1,870 @@
+import axios from "axios"
+import { IFormationCatalogue } from "../common/model/schema/formationCatalogue/formationCatalogue.types.js"
+import crypto from "crypto"
+import { getElasticInstance } from "../common/esClient/index.js"
+import { roundDistance } from "../common/geolib.js"
+import { FormationCatalogue } from "../common/model/index.js"
+import { IApiError, manageApiError } from "../common/utils/errorManager.js"
+import { regionCodeToDepartmentList } from "../common/utils/regionInseeCodes.js"
+import { trackApiCall } from "../common/utils/sendTrackingEvent.js"
+import { sentryCaptureException } from "../common/utils/sentryUtils.js"
+import { notifyToSlack } from "../common/utils/slackUtils.js"
+import config from "../config.js"
+import { formationMock, formationsMock, formationDetailMock } from "../mocks/formations-mock.js"
+import { formationsQueryValidator, formationsRegionQueryValidator } from "../service/formationsQueryValidator.js"
+import { IFormationEsResult } from "./formation.service.types.js"
+import { ILbaItem, ILbaItemTrainingSession, LbaItem } from "./shared.service.types.js"
+
+const formationResultLimit = 500
+
+const lbfDescriptionUrl = "https://labonneformation.pole-emploi.fr/api/v1/detail"
+
+const esClient = getElasticInstance()
+
+const diplomaMap = {
+  3: "3 (CAP...)",
+  4: "4 (BAC...)",
+  5: "5 (BTS, DEUST...)",
+  6: "6 (Licence, BUT...)",
+  7: "7 (Master, titre ingénieur...)",
+}
+
+const getDiplomaIndexName = (value) => {
+  if (value) {
+    return diplomaMap[value[0]]
+  } else {
+    return ""
+  }
+}
+
+/**
+ * Récupère les formations matchant les critères en paramètre depuis Elasticsearch
+ * @param {string[]} romes un tableau de codes ROME
+ * @param {string} romeDomain un domaine de ROME
+ * @param {[number, number]} coords des coordonnées au format [longitude,latitude] pour centrer la recherche
+ * @param {number} radius le rayon de recherche
+ * @param {string} diploma le niveau de diplôme visé
+ * @param {number} limit le nombre de résultats max à produire
+ * @param {string} caller l'identifiant fourni par l'exploitant de l'api
+ * @param {string} api le nom de l'api liée à cette fonction
+ * @param {string[]} options un tableau d'options modulant la recherche
+ * @param {string} useMock un flag indiquant s'il faut retourner des données mockées ou non
+ * @returns {Promise<IFormationEsResult[]>}
+ */
+export const getFormations = async ({
+  romes,
+  romeDomain,
+  coords,
+  radius,
+  diploma,
+  limit,
+  caller,
+  api = "formationV1",
+  options,
+  useMock,
+}: {
+  romes?: string[]
+  romeDomain?: string
+  coords?: [string, string]
+  radius?: number
+  diploma?: string
+  limit?: number
+  caller?: string
+  api?: string
+  options: string[]
+  useMock: string
+}): Promise<IFormationEsResult[]> => {
+  try {
+    if (useMock && useMock !== "false") {
+      return formationsMock
+    }
+
+    const distance = radius || 30
```

Comment:
```suggestion
    const distance = radius ?? 30
```

préférer ?? au lieu de || pour setter une valeur. https://github.com/tc39/proposal-nullish-coalescing#overview-and-motivation

Reply:
ok mais je vais d'abord m'assurer que le radius n'est pas parfois initialisé à 0 ou "" par erreur.

Reply:
bon en l'état radius est autorisé à 0 mais pas à "".  Or si quelqu'un mets 0 dans le call ou le remplace par 30 pour les formations pour forcer des résultats. (décision métier) L'illusion de la recherche en lieu exact.

Reply:
ok, faudrait typer le radius en number et passer 0 ou null pour 0 pour éviter le "" dans une intégration avec Joi